driver_test.go

```
@@ -12,26 +12,220 @@
 )
 
-func ExampleDriver() {
-	driver := &spannerdriver.Driver{
-		Options: []option.ClientOption{
-			option.WithCredentialsFile("/path/to/service-account-key.json"),
-		},
+var (
+	project  string
+	instance string
+	dbname   string
+	dsn      string
+	projset  bool
+	instset  bool
+	dbset    bool
+)
+
+type Connector struct {
+	ctx         context.Context
+	client      *spanner.Client
+	adminClient *adminapi.DatabaseAdminClient
+}
+
+func NewConnector() (*Connector, error) {
+
+	ctx := context.Background()
+
+	adminClient, err := adminapi.NewDatabaseAdminClient(
+		ctx,
+		option.WithoutAuthentication(),
+		option.WithEndpoint("0.0.0.0:9010"),
+		option.WithGRPCDialOption(grpc.WithInsecure()))
+	if err != nil {
+		return nil, err
+	}
+
+	dataClient, err := spanner.NewClient(ctx, dsn)
+	if err != nil {
+		return nil, err
+	}
+
+	curs := &Connector{
+		ctx:         ctx,
+		client:      dataClient,
+		adminClient: adminClient,
+	}
+	return curs, nil
+}
+
+func (c *Connector) Close() {
+	c.client.Close()
+	c.adminClient.Close()
+}
+
+// structs for row data
+type testaRow struct {
+	A string
+	B string
+	C string
+}
+type typeTestaRow struct {
+	stringt string
+	intt    int
+	floatt  float64
+	boolt   bool
+}
+
+func init() {
+
+	// get environment variables or set to default
+	if instance, instset = os.LookupEnv("SPANNER_TEST_INSTANCE"); !instset {
+		instance = "test-instance"
+	}
+	if project, projset = os.LookupEnv("SPANNER_TEST_PROJECT"); !projset {
+		project = "test-project"
+	}
+	if dbname, dbset = os.LookupEnv("SPANNER_TEST_DBNAME"); !dbset {
+		dbname = "gotest"
+	}
+
+	// derive data source name
+	dsn = "projects/" + project + "/instances/" + instance + "/databases/" + dbname
+}
+
+// Executes DDL statements
+func executeDdlApi(curs *Connector, ddls []string) {
+
+	op, err := curs.adminClient.UpdateDatabaseDdl(curs.ctx, &adminpb.UpdateDatabaseDdlRequest{
+		Database:   dsn,
+		Statements: ddls,
+	})
+	if err != nil {
+		log.Fatal(err)
```

Are we swallowing the error here? I think this should be returned to the caller.

---

fixed